libgalois/src/MemoryPolicy.cpp

```diff
@@ -67,61 +74,58 @@ katana::MemoryPolicyPerformance::MemoryPressureHigh(
 
 count_t
 katana::MemoryPolicyPerformance::ReclaimForMemoryPressure(
-    count_t active, count_t standby) const {
+    count_t standby) const {
   MemInfo mem_info;
-  UpdateMemInfo(&mem_info, active, standby);
+  UpdateMemInfo(&mem_info, standby);
+  count_t reclaim = standby;
 
-  if (mem_info.oom_score < 700 ||
-      mem_info.used_ratio < high_used_ratio_threshold()) {
+  if (mem_info.oom_score < 1000 ||
+      mem_info.used_ratio < high_used_ratio_threshold() ||
+      mem_info.available_bytes > 0.1 * physical()) {
     return 0;
   }
-  if (mem_info.oom_score < 850) {
-    LogIt(
-        fmt::format(
-            "reclaim for memory pressure {} GB", katana::ToGB(standby / 4)),
-        &mem_info);
-    return standby / 4;
-  }
+  // TODO (witchel) this might give back memory too quickly
   if (mem_info.oom_score < 1000) {
-    LogIt(
-        fmt::format(
-            "reclaim for memory pressure {} GB", katana::ToGB(standby / 2)),
-        &mem_info);
-    return standby / 2;
+    reclaim = standby / 4;
+  } else if (mem_info.oom_score < 1200) {
+    reclaim = standby / 2;
   }
   LogIt(
-      fmt::format("reclaim for memory pressure {} GB", katana::ToGB(standby)),
+      fmt::format("reclaim for memory pressure {} GB", katana::ToGB(reclaim)),
       &mem_info);
-  return standby;
+  return reclaim;
 }
 
 bool
 katana::MemoryPolicyPerformance::KillSelfForLackOfMemory(
-    count_t active, count_t standby) const {
+    count_t standby) const {
   MemInfo mem_info;
-  UpdateMemInfo(&mem_info, active, standby);
+  UpdateMemInfo(&mem_info, standby);
 
-  if (mem_info.oom_score > 1200 ||
-      mem_info.used_ratio > kill_used_ratio_threshold()) {
+  if ((mem_info.oom_score > kill_self_oom_threshold() ||
+       mem_info.used_ratio > kill_used_ratio_threshold()) &&
+      mem_info.available_bytes < 0.1 * physical()) {
     LogIt("KILL SELF", &mem_info);
     return true;
   }
   return false;
 }
 
 katana::MemoryPolicyPerformance::MemoryPolicyPerformance()
-    : MemoryPolicy(
-          {.high_used_ratio_threshold = 0.85,
-           .kill_used_ratio_threshold = 0.93}) {}
+    : MemoryPolicy({
+          .high_used_ratio_threshold = 0.85,
+          .kill_used_ratio_threshold = 0.95,
+          .kill_self_oom_threshold = 1280,
+          .high_pressure_oom_threshold = 1100,
```

Comment:
hold on, is it even possible to have a OOM score that is larger than 1000? If the score is usage * 10, wouldn't that be the max? Or can usage exceed 100% when we consider swap as well?

Reply:
It can be greater than 1000.  I don't know what the scale is.

Reply:
Interesting. Some sources I read online say that the max and others say that it can be greater than 1000. I wonder if it's based on the kernel version.

Reply:
I am worried about being robust across different kernel versions.  This code has changed a bunch in 10+ years.  There is also `/proc/self/oom_adj` (deprecated) and `/proc/self/oom_score_adj`.  I don't deal with these right now.